core/common/src/main/scala/org/typelevel/otel4s/baggage/BaggageManager.scala

```diff
@@ -30,29 +33,55 @@ import cats.mtl.Local
   *     _  <- Console[F].println("user_id: " + id)
   *   } yield ()
   *
-  * BaggageManager[F].local(io)(_.updated("user_id", "uid"))
+  * BaggageManager[F].modifiedScope(_.updated("user_id", "uid"))(io)
   *   }}}
   */
-trait BaggageManager[F[_]] extends Local[F, Baggage] {
+trait BaggageManager[F[_]] {
+
+  // TODO: remove after finishing migrating away from Local
+  protected[BaggageManager] def applicative: Applicative[F]
 
-  /** @return the current `Baggage` */
-  final def current: F[Baggage] = ask[Baggage]
+  /** @return the current scope's `Baggage` */
+  def current: F[Baggage]
 
   /** @return
     *   the [[Baggage.Entry entry]] to which the specified key is mapped, or `None` if the current `Baggage` contains no
     *   mapping for the key
     */
-  def get(key: String): F[Option[Baggage.Entry]] =
-    reader(_.get(key))
+  def get(key: String): F[Option[Baggage.Entry]]
 
   /** @return
     *   the value (without [[Baggage.Metadata metadata]]) to which the specified key is mapped, or `None` if the current
     *   `Baggage` contains no mapping for the key
     */
-  def getValue(key: String): F[Option[String]] =
-    reader(_.get(key).map(_.value))
+  def getValue(key: String): F[Option[String]]
+
+  /** Creates a new scope in which a modified version of the current `Baggage` is used. */
+  def modifiedScope[A](modify: Baggage => Baggage)(fa: F[A]): F[A]
+
+  /** Creates a new scope in which the given `Baggage` is used. */
+  def replacedScope[A](baggage: Baggage)(fa: F[A]): F[A]
 }
 
 object BaggageManager {
   def apply[F[_]](implicit ev: BaggageManager[F]): BaggageManager[F] = ev
+
+  private[this] def asLocal[F[_]](bm: BaggageManager[F]): Local[F, Baggage] =
+    new Local[F, Baggage] {
+      def applicative: Applicative[F] = bm.applicative
+      def ask[E2 >: Baggage]: F[E2] = applicative.widen(bm.current)
+      def local[A](fa: F[A])(f: Baggage => Baggage): F[A] =
+        bm.modifiedScope(f)(fa)
+    }
+
+  @deprecated(
+    "BaggageManager no longer extends Local. Use `current`, `modifiedScope` and `replacedScope`",
+    since = "1.13.0"
+  )
+  implicit def bmAsExplicitLocal[F[_]](bm: BaggageManager[F]): Local[F, Baggage] =
+    asLocal(bm)
+
+  @deprecated("BaggageManager no longer extends Local", since = "1.13.0")
```

Comment:
Should it be `0.13.0`?
```suggestion
  @deprecated("BaggageManager no longer extends Local", since = "0.13.0")
```